### PR TITLE
 Update Supported Kubernetes to 1.27, 1.28, 1.29

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Work on #3623 

**Special notes for your reviewer**:

### CODE REVIEW

File: build/terraform/e2e/module.tf [hunk 1]
```diff
@@ -38,9 +38,9 @@ variable "kubernetes_versions" {
   description = "Create e2e test clusters with these k8s versions in these regions"
   type        = map(list(string))
   default     = {
-    "1.26" = ["asia-east1", "REGULAR"]
     "1.27" = ["us-east1", "REGULAR"]
```
Comment (igooch) [inline build/terraform/e2e/module.tf:41]: 1.27 and 1.28 should be updated to "RAPID"

**Comment by Kalaiselvi84** [comment]
@igooch,  Just to let you know, the only change from last time is that I didn't include the patch version of AKS as per Max's suggestion. Will that make a difference?

**Comment by igooch** [comment]
> @igooch, Just to let you know, the only change from last time is that I didn't include the patch version of AKS as per Max's suggestion. Will that make a difference?

It's the GKE Autopilot 1.29 that's failing, and it fails locally for me as well. So it's a GKE autopilot specific issue, although I haven't yet been able to track it down.

File: cloudbuild.yaml [hunk 2]
```diff
@@ -285,7 +285,7 @@ steps:
         set -o pipefail
         pids=()
         cloudProducts=("generic" "gke-autopilot")
-        declare -A versionsAndRegions=( [1.26]=asia-east1 [1.27]=us-east1 [1.28]=us-west1 )
+        declare -A versionsAndRegions=( [1.27]=us-east1 [1.28]=us-west1 [1.29]=us-central1 )
```
Comment (igooch) [inline cloudbuild.yaml:288]:
```suggestion
        declare -A versionsAndRegions=( [1.27]=us-east1 [1.28]=us-west1 [1.29]=asia-east1 )
```

Comment (Kalaiselvi84) [reply cloudbuild.yaml]: changed the region to asia-east1 for 1.29

**Comment by Kalaiselvi84** [comment]
shall we try redoing the process?

**Comment by igooch** [comment]
> shall we try redoing the process?

Yes, could you re-run the most recent commit? The test clusters were in the middle of rebuilding.

```
git commit --amend --date=now
git push --force-with-lease --force-if-includes
```

**Comment by Kalaiselvi84** [comment]
> Yes, could you re-run the most recent commit? The test clusters were in the middle of rebuilding.
> 
> ```
> git commit --amend --date=now
> git push --force-with-lease --force-if-includes
> ```

Rebuilding [this PR](https://github.com/googleforgames/agones/pull/3657) manually.

**Comment by markmandel** [comment]
Looks like 1.29 standard worked, but autopilot failed. Retrying to see what happens.

**Comment by roberthbailey** [comment]
We are going to pause on this until after the next release (and likely until after GDC) to make sure everything works properly on Kubernetes 1.29.

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**